### ISSUE TITLE
Rippled response causes failed forward

### ISSUE THREAD

**Issue by cindyyan317** [XRPLF/clio#777]
UPDATE: we should remove `full` entirely as it's now deprecated. let's also remove `accounts` and any other deprecated fields. See the [doc here](https://xrpl.org/ledger.html#request-format).

-------
Rippled responds slowly for the request 
```
{
    "method": "ledger",
    "params": [
        {
            "ledger_index": "validated",
            "accounts": false,
            "full": true,
            "transactions": false,
            "expand": false,
            "owner_funds": false
        }
    ]
}
```
which causes failed forward. 
Do we need to increase the timeout or give warning for this scenario? @godexsoft

**Comment by godexsoft** [comment]
I tried to increase the timeout from 3 to 10 seconds to only discover that the RPC call you shared above takes **forever** and fails to complete with default curl command when I go directly to rippled (after 5 and a half minutes): 
``` 
curl: (18) transfer closed with 6499724125 bytes remaining to read
parse error: Unfinished string at EOF at line 1, column 339695397
```

I don't think we can do anything with timeouts in this case for forwarding.

**Comment by godexsoft** [comment]
@ximinez @Bronek Any recommendation from your side on this one? 🙇

**Comment by ximinez** [comment]
From https://xrpl.org/ledger.html#request-format

> The `ledger` field is deprecated and may be removed without further notice. The `full`, `accounts`, and `type` fields (admin-only) are also deprecated; the Clio server does not implement these parameters.

The ledger (on Mainnet) is so big that there is no reasonable way to build the full json response before the connection times out, or rippled runs out of memory, or whatever. For that reason, even though `full` still exists, it is restricted to admin connections - the idea is that it might be useful for testing, or for someone to run on their own server for their own purposes, but it is strongly discouraged.

`ledger_data` is the replacement, which uses paging markers so that it's not trying to build the whole thing at once.

tl;dr - It's a known issue. Don't try to use it on mainnet.

**Comment by ximinez** [comment]
The previous version of that page included:

> | `full`         | Boolean          | No        | **Admin only** If `true`, return full information on the entire ledger. Ignored if you did not specify a ledger version. The default is `false`. (Equivalent to enabling `transactions`, `accounts`, and `expand`.) The [Clio server](the-clio-server.html) does not support this field. **Caution:** On Mainnet, this can be gigabytes worth of data, so the request is likely to time out. |

**Comment by godexsoft** [comment]
@ximinez If it's deprecated - that's great. We should remove all mentions of it from Clio and be done. Currently (and in 2.1.0 that just got released), specifying `full` for the request will forward to `rippled`. We should remove this by next release. Thanks 🙇

**Comment by ximinez** [comment]
@godexsoft You're welcome!

Though I am curious if `clio` would have any more success building the `full` ledger locally. I don't expect it will be, because despite its efficiencies, we're still talking about gigs of json being built in memory.

**Comment by godexsoft** [comment]
Please also check that forwarding got some timeout setting.

**Comment by godexsoft** [comment]
> Please also check that forwarding got some timeout setting.

30s is the default timeout for reading data by the forwarding proxy. Connection timeout is set separately and it is 3s.
